### ISSUE TITLE
NGINX Configuration

### ISSUE THREAD

**Issue by elwali10** [wazuh/wazuh-documentation#1016]
With the current configuration for NGINX ( https://documentation.wazuh.com/current/installation-guide/optional-configurations/kibana_ssl.html) :

```
server {
    listen 80;
    listen [::]:80;
    return 301 https://$host$request_uri;
}

server {
    listen 443 default_server;
    listen            [::]:443;
    ssl on;
    ssl_certificate /etc/pki/tls/certs/kibana-access.pem;
    ssl_certificate_key /etc/pki/tls/private/kibana-access.key;
    access_log            /var/log/nginx/nginx.access.log;
    error_log            /var/log/nginx/nginx.error.log;
    location / {
        auth_basic "Restricted";
        auth_basic_user_file /etc/nginx/conf.d/kibana.htpasswd;
        proxy_pass http://kibana-server-ip:5601/;
    }
}
```
We are able to access to kibana using https but also using http://kibana-ip:5601 and that's not the attended behaviour. adding the following configuration fixed it and block access using http :

```
server {
    listen 80;
    listen [::]:80;
    return 301 https://$host$request_uri;
}

server {
    listen 443 default_server;
    listen            [::]:443;
    ssl on;
    ssl_certificate /etc/pki/tls/certs/kibana-access.pem;
    ssl_certificate_key /etc/pki/tls/private/kibana-access.key;
    access_log            /var/log/nginx/access.log;
    error_log            /var/log/nginx/error.log;
    auth_basic "Restricted";
    auth_basic_user_file /etc/nginx/conf.d/kibana.htpasswd;
    location / {
        proxy_pass http://localhost:5601/;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
}
```

Another thing to mention we have two pages related to NGinx configuration with more or less different content  :

- https://documentation.wazuh.com/current/learning-wazuh/build-lab/kibana-setup.html?highlight=nginx

- https://documentation.wazuh.com/current/installation-guide/optional-configurations/kibana_ssl.html?highlight=nginx

**Comment by JaoChaos** [comment]
Hi @elwali10,

thanks for reporting this.
Actually, I'm working on this issue: https://github.com/wazuh/wazuh-documentation/issues/734

So I'm going to add this fix into that PR :+1: 

Best regards.

**Comment by jesusgn90** [comment]
Hi @elwali10,

After researching this ticket I've concluded that the documentation is just fine, the only thing I've changed has been https://github.com/wazuh/wazuh-documentation/commit/dbb0847242f65688e83c467a0e546068f2e353cb, replacing `kibana-server-ip` with `localhost`.

If your Kibana is just exposed to `localhost` (`server.host: "localhost"`) the exposed service is NGINX, which redirects from HTTP to HTTPS and adds basic auth, then it sends incoming requests to `localhost`:5601 which is the local service for Kibana. There is no way to access IP:5601 if you are not on `localhost`.

If you have `server.host` as `0.0.0.0` for Kibana, NGINX is working properly but you are also exposing the Kibana server. Let me know if this worked for you, and re-open the ticket if you have any problem with this resolution.

Regards
